mlx/compile.cpp

```diff
@@ -415,6 +415,11 @@ void compile_simplify(
   }
   tape = std::move(new_tape);
 
+  std::map<std::uintptr_t, uint32_t> tape_order;
```

Comment:
Out of curiosity why a `map` instead of an `unordered_map`?

Reply:
Typo!